kayenta-core/src/main/java/com/netflix/kayenta/canary/CanaryConfig.java

```diff
@@ -39,6 +39,8 @@
 
   @NotNull @Getter @Setter private String name;
 
+  @NotNull @Getter private String id;
```

Comment:
If this field is not null, then you should populate this field on the GET requests to configs that already exist but do not yet have the id saved as part of the object.

https://github.com/spinnaker/kayenta/blob/c2acf40a2752d8748ce466325f09445321ce6299/kayenta-web/src/main/java/com/netflix/kayenta/controllers/CanaryConfigController.java#L61-L79

Reply:
Maybe this should be @nullable, so that way if JSR 308 validation is turned on people who don't supply an id, since it's technically optional won't get an API error.

Reply:
Good point. Have removed `@NotNull`.